### ISSUE TITLE
common api for getting gradient from all backend?

### ISSUE THREAD

**Issue by pure-rgb** [keras-team/keras#20760]
In this example, https://keras.io/examples/vision/grad_cam/, it is used as follows to get gradient using tensorflow.

```python
with tf.GradientTape() as tape:
   last_conv_layer_output, preds = grad_model(img_array)
   if pred_index is None:
      pred_index = tf.argmax(preds[0])
   class_channel = preds[:, pred_index]
```

Changing the backend won't work without chaning the way to getting gradient, so I was wondering if keras care about providing general api interface to get the gradient, etc.

**Comment by harshaljanjani** [comment]
Thanks for posting the issue @pure-rgb! 
For now, the state of the library is such that each backend uses its native gradient computation system when working with training loops. In principle, Keras does aim to provide a unified high-level API for many operations, but gradient computation touches core autograd mechanisms that are pretty different between frameworks, and unifying them, imo would add unnecessary complexity and potential maintenance overhead. The documentation has example scripts showing the idiomatic way to work with gradients for each supported backend (such as `writing_a_custom_training_loop_in_jax` and alternatives for `torch` and `tensorflow`), which provide a useful enough set of insights with well-elaborated docstrings. In summary of the alternatives:

For PyTorch backend:
```python
output.backward()
gradients = input.grad
```
For JAX backend:
```python
gradients = jax.grad(loss_fn)(params)
```

Design Choices:
- TensorFlow: Explicit gradient tracking (`tf.GradientTape`)
- PyTorch: Automatic gradient accumulation (`loss.backward()`)
- JAX: Functional approach with pure functions

While I'm not aware of whether it's in consideration, it seems highly gratuitous to me, but I'm sure the collaborators could provide more insights into it!

**Comment by pure-rgb** [comment]
@dhantule 
Understood.
